sdk/rust/src/ledger_id.rs

```diff
@@ -26,6 +32,10 @@ impl LedgerId {
         Self(vec![2])
     }
 
+    pub fn from_bytes(bytes: Vec<u8>) -> Self {
+        Self(bytes)
+    }
+
     #[must_use]
     pub fn is_mainnet(&self) -> bool {
```

Comment:
(not to this line, and maybe a future PR type thing, I can make an issue if so)
it'd be nice if there was a `pub(crate)` way of getting the network (motivation: more understandable code? Subjectively)

```rs
// don't really need `Eq` or `PartialEq`
#[derive(Copy, Clone)]
pub(crate) enum KnownNetwork {
    Mainnet,
    Testnet,
    Previewnet,
}

impl KnownNetwork {
    #[must_use]
    pub(crate) const fn as_str(self) -> &'static str {
        todo!()
    }
}

impl LedgerId {
    #[must_use]
    pub(crate) const fn network(&self) -> Option<KnownNetwork> {
        match &self.0 {
             [0] => Some(KnownNetwork::Mainnet),
             // ...
    }

    #[must_use]
    pub fn is_mainnet(&self) -> bool {
        matches!(self.network(), Some(KnownNetwork::Mainnet)
    }

    // ... other `is_*` functions ...

    #[must_use]
    pub fn is_known_network(&self) -> bool {
        self.network().is_some()
    }
}

impl Display for LedgerId {
    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
        // can use `self.network()`
    }
```

Reply:
We had something similar to this in the Java SDK called `NetworkName` (terrible name, I know), and we deprecated it in favor of `LedgerId` and it's not used anywhere outside of deprecated functions.  From my experience there, I don't think we need this.

Reply:
fair, I meant as an internal only thing, to be clear, but if you don't think it'd work that's fine